numba/cuda/compiler.py

```diff
@@ -272,8 +273,9 @@ def compile_ptx(pyfunc, args, debug=False, lineinfo=False, device=False,
         lib = cres.library
     else:
         tgt = cres.target_context
-        filename = cres.type_annotation.filename
-        linenum = int(cres.type_annotation.linenum)
+        func_identity = FunctionIdentity.from_function(pyfunc)
+        filename = func_identity.filename
+        linenum = func_identity.firstlineno
```

Comment:
Could we just do this similarly to how  `_Kernel` does it instead of introducing a dependency between the `bytecode` module and the CUDA `compiler` module,

https://github.com/numba/numba/blob/05e5e3407ba0ff3fb454943e03f6d76255556cbe/numba/cuda/dispatcher.py#L82-L84

 i.e.:

```suggestion
        code = pyfunc.__code__
        filename = code.co_filename
        linenum = code.co_firstlineno
```
?

Reply:
@gmarkall PR updated as requested.